src/Http/WebUtilities/src/HttpResponseStreamWriter.cs

```diff
@@ -136,7 +136,7 @@ public override void Write(ReadOnlySpan<char> value)
             }
 
             var written = 0;
-            while (written < value.Length)
+            while (true)
```

Comment:
I like keeping the terminating condition in the loop condition. Right now, it looks like `written` should be cumulative since it's defined outside of the loop, but really it's just that size of the last chunk that was written.

I would do `var remaining = value.length;` instead of `var written = 0;`, and then make `while (remaining > 0)` the loop condition. Then you can just put `remaining -= written` after `var written = CopyToCharBuffer(value);` and keep everything else the same as it was. I think that makes it much easier to see what's going on.